Advanced Filters: Ranges: Allow update only on complete selection

psealock opened the pull request:
Fixes #

Advanced Filters enables the "Filter" button when a filter has all values defined AND it will produce a new url that is different than the current one.

This PR updates the logic for "between" values (Numerical or Date. Although testing with Date ranges won't work until #1305 gets merged). You can only update the filters if both max and min (or before/after) values are set.

### Screenshots

![screen shot 2019-01-17 at 12 50 26 pm](https://user-images.githubusercontent.com/1922453/51286155-a0df2900-1a56-11e9-916b-a687599aacdd.png)

### Detailed test instructions:

1. Customers Report > Show > Advanced Filters > Add a filter > No. of Orders.
2. See the "Filter" button disabled.
3. Add a minimum value.
4. See the "Filter" button disabled.
5. Add a maximum value.
6. See the "Filter" button enabled.

psealock committed:
Advanced Filters: Ranges: Allow update only on complete selection

joshuatf commented on packages/navigation/src/test/filters.js:
```diff
@@ -126,6 +126,18 @@ describe( 'getQueryFromActiveFilters', () => {
 		expect( nextQuery.with_search_includes ).toBeUndefined();
 	} );
+
+	it( 'should only reflect complete filters with multiple values', () => {
```
Thanks for this test 👍

joshuatf reviewed (commented):
Looks good and is testing well.

One thing I noticed was that if you use another match type (i.e., less than or more than) and enter a value to enable a filter, the filter is still enabled if you switch to `between` and allow you to submit a bad value.

Not sure if that's something we want to address here or another PR.

psealock commented:
Nice find @joshuatf ! I hadn't factored in that scenario. I added an explicit check for the `between` rule instead of just the presence of an array for the value. I also updated the test to reflect this case.

jeffstieler reviewed (approved):
Code looks good and tests well! :shipit:
